### PR TITLE
timer: timer unit must depend on service unit.

### PR DESCRIPTION
**Pull request by olifre**
#### Pull Request (PR) description
Without this, issues like:
```
cvmfs-fsck.timer: Refusing to start, unit cvmfs-fsck.service to trigger not loaded.
```
may be observed.

### CODE REVIEW

**Comment by kenyon** [comment]
Should we update https://github.com/voxpupuli/puppet-systemd/blob/master/spec/defines/timer_spec.rb with this too?

**Comment by olifre** [comment]
That seems like a good idea, I'll add a `that_comes_before` there ;-).

**Comment by olifre** [comment]
@kenyon I'm not sure why the `spec` tests fail — can you give me a pointer?

**Comment by bastelfreak** [comment]
mhm, odd that Inuwed this pattern a lot in the past and never run into the error

File: spec/defines/timer_spec.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@
           it {
             expect(subject).to contain_systemd__unit_file('foobar.service').with(
               content: "[Service]\nExecStart=/bin/touch /tmp/foobar"
-            )
+            ).that_comes_before('Systemd::Unit_File[\'foobar.timer\']')
```
Comment (ekohl) [inline spec/defines/timer_spec.rb:32]: No need to quote this. I if you apply this everywhere the tests will pass
```suggestion
            ).that_comes_before('Systemd::Unit_File[foobar.timer]')
```

Comment (olifre) [reply spec/defines/timer_spec.rb]: Many thanks for pointing this out! That was unexpected, I thought the syntax would be similar to that used in Puppet directly. 
Fixed and force-pushed :-).

**Review by ekohl — APPROVED** [review]
This is actually correct in that systemd requires it. It's probably that the definition order generally speaking is "good enough". This is explicit and thus better. :+1: if the tests are green.

File: spec/defines/timer_spec.rb (hunk 1 above)
Comment (olifre) [reply spec/defines/timer_spec.rb]: It seems tests still did not like this, I changed:
`Systemd::Unit_File` => `Systemd::Unit_file` now, assuming capitalization needs to match explicitly. Let's see if that makes things more green.

Comment (olifre) [reply spec/defines/timer_spec.rb]: That did the trick :smile:.

**Review by kenyon — APPROVED** [review]

**Pull request merged**